lib/spack/spack/build_environment.py

```diff
@@ -323,6 +336,20 @@ def set_build_environment_variables(pkg, env, dirty):
     link_deps       = set(pkg.spec.traverse(root=False, deptype=('link')))
     build_link_deps = build_deps | link_deps
     rpath_deps      = get_rpath_deps(pkg)
+    build_run_deps = set()
```

Comment:
Can we call this `run_deps_of_build_deps` or something similar? `build_run_deps` may be confused with `pkg.spec.traverse(root=False, deptype=('build', 'run'))`.

Reply:
I have renamed it to `build_and_supporting_deps` and added a clarifying comment